From 8c2b4061d16756ee2ed37f08db063b8215656943 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Thu, 2 Nov 2017 16:48:15 +0800
Subject: [PATCH] fix configure failed while build dir name contains "yes"

While build dir name contained "yes", and compiled with
producing debugging information (-g), the configure failed.

The pattern of AC_EGREP_CPP (string "yes") mismatched with
dir name. It caused the test of AC_EGREP_CPP incorrect.

Use YES_IS_DEFINED to replace yes which is not often used
as dir name.

Bug: 52329
(Fix release v.4.6.2 or above)

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 configure.ac | 32 ++++++++++++++++----------------
 1 file changed, 16 insertions(+), 16 deletions(-)

diff --git a/src/configure.ac b/src/configure.ac
index 28237f5..34f9735 100644
--- a/configure.ac
+++ b/configure.ac
@@ -151,9 +151,9 @@ fi
 
 
 AC_CHECKING(for Ultrix)
-AC_EGREP_CPP(yes,
+AC_EGREP_CPP(YES_IS_DEFINED,
 [#if defined(ultrix) || defined(__ultrix)
-   yes;
+   YES_IS_DEFINED;
 #endif
 ], ULTRIX=1)
 
@@ -168,9 +168,9 @@ dnl ghazi@caip.rutgers.edu (Kaveh R. Ghazi):
 dnl BBN butterfly is not POSIX, but a MACH BSD system.
 dnl Do not define POSIX and TERMIO.
 AC_CHECKING(for butterfly)
-AC_EGREP_CPP(yes,
+AC_EGREP_CPP(YES_IS_DEFINED,
 [#if defined(butterfly)
-  yes;
+  YES_IS_DEFINED;
 #endif
 ], butterfly=1)
 
@@ -179,12 +179,12 @@ if test -n "$ULTRIX"; then
   test -z "$GCC" && CC="$CC -YBSD"
 fi
 AC_CHECKING(for POSIX.1)
-AC_EGREP_CPP(yes,
+AC_EGREP_CPP(YES_IS_DEFINED,
 [#include <sys/types.h>
 #include <unistd.h>
 main () {
 #ifdef _POSIX_VERSION
-  yes;
+  YES_IS_DEFINED;
 #endif
 ], AC_NOTE(- you have a POSIX system) AC_DEFINE(POSIX) posix=1)
 fi
@@ -196,9 +196,9 @@ AC_TRY_COMPILE(
 #include <fcntl.h>], [int x = SIGCHLD | FNDELAY;], , AC_DEFINE(SYSV))
 
 AC_CHECKING(for sequent/ptx)
-AC_EGREP_CPP(yes,
+AC_EGREP_CPP(YES_IS_DEFINED,
 [#ifdef _SEQUENT_
-  yes;
+  YES_IS_DEFINED;
 #endif
 ], LIBS="$LIBS -lsocket -linet";seqptx=1)
 
@@ -213,9 +213,9 @@ AC_TRY_LINK([#include <utmpx.h>
 AC_CHECK_HEADERS([stropts.h string.h strings.h])
 
 AC_CHECKING(for Solaris 2.x)
-AC_EGREP_CPP(yes,
+AC_EGREP_CPP(YES_IS_DEFINED,
 [#if defined(SVR4) && defined(sun)
-  yes
+  YES_IS_DEFINED;
 #endif
 ], LIBS="$LIBS -lsocket -lnsl -lkstat")
 
@@ -699,9 +699,9 @@ else
 pdir='/dev'
 fi
 dnl SCO uses ptyp%d
-AC_EGREP_CPP(yes,
+AC_EGREP_CPP(YES_IS_DEFINED,
 [#ifdef M_UNIX
-   yes;
+   YES_IS_DEFINED;
 #endif
 ], ptys=`echo /dev/ptyp??`, ptys=`echo $pdir/pty??`)
 dnl if test -c /dev/ptyp19; then
@@ -878,9 +878,9 @@ fi
 )
 
 if test -z "$load" ; then
-AC_EGREP_CPP(yes,
+AC_EGREP_CPP(YES_IS_DEFINED,
 [#if defined(NeXT) || defined(apollo) || defined(linux)
-  yes;
+  YES_IS_DEFINED;
 #endif
 ], load=1)
 fi
@@ -1105,9 +1105,9 @@ AC_CHECKING(syslog in libbsd.a)
 AC_TRY_LINK(, [closelog();], AC_NOTE(- found.), [LIBS="$oldlibs"
 AC_NOTE(- bad news: syslog missing.) AC_DEFINE(NOSYSLOG)])])
 
-AC_EGREP_CPP(yes,
+AC_EGREP_CPP(YES_IS_DEFINED,
 [#ifdef M_UNIX
-   yes;
+   YES_IS_DEFINED;
 #endif
 ], LIBS="$LIBS -lsocket -lcrypt_i")
 
-- 
2.17.1

